src/Artifacts.php

```diff
@@ -39,26 +42,31 @@ public function getFilesystem(): Filesystem
         return $this->filesystem;
     }
 
-    public function upload(Tags $tags): array
+    public function upload(Tags $tags, array $configuration = []): array
     {
         if (!$this->checkConfigId($tags)) {
             return [];
         }
 
-        $uploaded = [];
-        $uploaded[] = $this->uploadArtifact(
+        $current = $this->uploadArtifacts(
             $this->filesystem->getUploadCurrentDir(),
-            $tags
+            $tags,
+            $configuration
         );
 
+        $shared = [];
         if ($tags->getOrchestrationId()) {
-            $uploaded[] = $this->uploadArtifact(
+            $shared = $this->uploadArtifacts(
                 $this->filesystem->getUploadSharedDir(),
-                $tags->setIsShared(true)
+                $tags->setIsShared(true),
+                $configuration
             );
         }
 
-        return array_filter($uploaded);
+        return [
+            'current' => $current,
+            'shared' => $shared,
+        ];
```

Comment:
Tady ten return sa malinko zmenil, ale v podstate len v tom, ze teraz tie polia maju "keys", predtym to boli dve polia so nazvami suborov.